### PR TITLE
Add URL routes to each recipe

### PR DESCRIPTION
**Pull request by cal-stytch**

### CODE REVIEW

File: lib/recipeData.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          signupExpirationMinutes: 30,
+          createUserAsPending: false,
+        },
+        oauthOptions: {
+          providers: [
+            { type: OAuthProvidersTypes.Google, one_tap: true, position: 'embedded' },
+            { type: OAuthProvidersTypes.Apple },
+            { type: OAuthProvidersTypes.Microsoft },
+            { type: OAuthProvidersTypes.Facebook },
+            { type: OAuthProvidersTypes.Github },
```
Comment (chris-stytch) [inline lib/recipeData.tsx:40]:
```suggestion
            { type: OAuthProvidersTypes.Github },
            { type: OAuthProvidersTypes.GitLab },
```
We missed hard coding that addition here.

File: lib/recipeData.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import { LoginType } from './types';
+import LoginWithSMS from '../components/SMSPasscodes/LoginWithSMS';
+import LoginWithEmailWebAuthn from '../components/EmailWebAuthn/LoginWithEmail';
+import LoginWithMagicLinks from '../components/LoginWithReactSDK';
+import LoginProducts from './loginProduct';
+
+export const Recipes: Record<string, LoginType> = {
+  REACT: {
+    id: 'ui-js-sdk',
```
Comment (chris-stytch) [inline lib/recipeData.tsx:9]:
```suggestion
    id: 'sdk-ui-oauth',
```

File: lib/recipeData.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            { type: OAuthProvidersTypes.Github },
+          ],
+          loginRedirectURL: REDIRECT_URL_BASE + '/authenticate?type=oauth',
+          signupRedirectURL: REDIRECT_URL_BASE + '/authenticate?type=oauth',
+        },
+      };
+      `,
+  },
+  CUSTOM_UI_HEADLESS: {
+    id: 'headless',
```
Comment (chris-stytch) [inline lib/recipeData.tsx:49]:
```suggestion
    id: 'sdk-sms',
```

File: lib/recipeData.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  // Verify the passcode input by the user
+  await stytchClient.otps.authenticate(
+    otpInput,
+    methodId,
+    { session_duration_minutes: 30 }
+  );
+      `,
+  },
+  CUSTOM_UI_API: {
+    id: 'api',
```
Comment (chris-stytch) [inline lib/recipeData.tsx:69]:
```suggestion
    id: 'api-webauthn',
```

File: lib/recipeData.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        login_magic_link_url:  REDIRECT_URL_BASE + '/api/authenticate_magic_link_with_webauthn',
+        signup_magic_link_url: REDIRECT_URL_BASE + '/api/authenticate_magic_link_with_webauthn',
+      });
+  
+      // Authenticate the Email magic link
+      await stytchClient.magicLinks.authenticate(token as string);
+      `,
+  },
+  CRYPTO_WALLETS: {
+    id: 'crypto-wallets',
```
Comment (chris-stytch) [inline lib/recipeData.tsx:90]:
```suggestion
    id: 'sdk-crypto-wallets',
```

**Review by chris-stytch — APPROVED** [review]
Renamed routes to be more explicitly based on integration type and product. I think we'll want to deep link based on product, but I definitely foresee us wanting to deep link based on type as well.

File: lib/recipeData.tsx (hunk 1 above)
Comment (cal-stytch) [reply lib/recipeData.tsx:40]: Will add!

**Pull request merged**